pdns/ixfrdist.cc

```diff
@@ -636,7 +636,8 @@ void handleTCPRequest(int fd, boost::any&) {
     }
   }
 
-  res = recv(cfd, &buf, sizeof(buf), 0);
+  size_t toRead = std::min(static_cast<size_t>(ntohs((buf[0]<<8) + buf[1])), sizeof(buf));
```

Comment:
Just a nit but it might be easier to read the size into a `uint16_t` than doing bits shifting